### PR TITLE
Update deps and fix vulnerabilities

### PR DESCRIPTION
**Pull request by niklaslong**
Updates a number of vulnerable dependencies. 

Firstly, `loom` which has the `generator` [vulnerability](https://rustsec.org/advisories/RUSTSEC-2020-0151), this should be fully fixed. It also now has the `chrono` [vulnerability](https://rustsec.org/advisories/RUSTSEC-2020-0159), though `loom` still needs a version bump, a simple `cargo update` will suffice once it's released. The [PR](https://github.com/tokio-rs/loom/issues/237) has been merged so it shouldn't be long now. 

I've also updated `tokio` and `async-std`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,14 +23,14 @@ std = []
...
+loom = "0.5"
 
 [dev-dependencies]
 criterion = "0.3"
 
 [target.'cfg(not(loom))'.dev-dependencies]
-tokio = { version = "0.2.20", features = ["rt-threaded", "macros", "time"] }
-async-std = { version = "1.5.0", features = ["attributes"] }
+tokio = { version = "1", features = ["rt", "macros", "time"] }
+async-std = { version = "1", features = ["attributes"] }
```
Comment (faern) [inline Cargo.toml:33]: Why is `async-std` changed downwards from 1.5 to 1?

**Review by faern — COMMENTED** [review]
This looks good overall! Thank you. One minor comment. And can you please rebase this on latest master so I can see that it passes the new github actions CI. The MSRV for loom has been bumped, so you will need to do the equivalent bump in `.github/workflows/build-and-test.yml`.

I'm also looking forward to the next `loom` release. They just implemented `park` and `unpark` directly in `loom`, so we don't have to hack around it here! https://github.com/tokio-rs/loom/pull/240

File: Cargo.toml (hunk 1 above)
Comment (niklaslong) [reply Cargo.toml:33]: Happy to reintroduce the minor version though this will default to the latest version with `cargo build`.

**Comment by niklaslong** [comment]
Rebased. @faern you may need to trigger the build manually depending on the repo permissions.

File: Cargo.toml (hunk 1 above)
Comment (faern) [reply Cargo.toml:33]: It's fine maybe.. Thinking about adding a CI job with `cargo update -Z minimal-versions` to make sure stuff like this does not break. But it's fine for now I guess.

**Review by faern — APPROVED** [review]

**Comment by faern** [comment]
Thanks for the contribution!

**Pull request merged**